In [16]:

import plotly.express as px

In [17]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [18]:
#%sql postgresql://postgres:Letsdoit!@localhost:5432/Eskom_db
%sql postgresql://postgres:postgres@localhost:5430/eskom_db

'Connected: postgres@eskom_db'

In [19]:
%%sql 
DROP TABLE IF EXISTS eskom_data;
CREATE TABLE  eskom_data(
    date_time_hour_beginning DATE,
    original_res_forecast_before_lockdown NUMERIC,
    residual_forecast  NUMERIC,	
    rsa_contracted_forecast	 NUMERIC,
    dispatchable_generation	NUMERIC ,
    residual_demand NUMERIC,
    rsa_contracted_demand  NUMERIC,
    international_exports  NUMERIC,
    international_imports  NUMERIC,
    thermal_generation  NUMERIC,
    nuclear_generation  NUMERIC,
    eskom_gas_generation  NUMERIC,	
    eskom_ocgt_generation  NUMERIC,	
    hydro_water_generation  NUMERIC,
    pumped_water_generation	 NUMERIC,
    ils_usage  NUMERIC,
    manual_load_reduction NUMERIC,
    ios_excl_ils_and_mlr  NUMERIC,
    dispatchable_ipp_ocgt  NUMERIC,
    eskom_gas_sco  NUMERIC,
    eskom_ocgt_sco  NUMERIC,
    hydro_water_sco	 NUMERIC,
    pumped_water_sco  NUMERIC,
    pumping	 NUMERIC,
    wind  NUMERIC,
    pv  NUMERIC,
    csp  NUMERIC,
    other_re  NUMERIC,	
    total_re  NUMERIC,	
    wind_installed_capacity	 NUMERIC,
    pv_installed_capacity  NUMERIC,	
    csp_installed_capacity  NUMERIC,
    other_re_installed_capacity	 NUMERIC,
    total_re_installed_capacity	 NUMERIC,
    installed_eskom_capacity  NUMERIC,
    total_pclf  NUMERIC,
    total_uclf  NUMERIC,
    total_oclf  NUMERIC,
    total_uclf_and_oclf	 NUMERIC,
    non_comm_sentout  NUMERIC,	
    drakensberg_gen_unit_hours  NUMERIC,
    palmiet_gen_unit_hours  NUMERIC,	
    ingula_gen_unit_hours NUMERIC
    
);

 * postgresql://postgres:***@localhost:5430/eskom_db
Done.
Done.


[]

In [ ]:
#FROM 'C:\Users\Public\ESK2033_clean.csv'

In [45]:
%%sql 
COPY eskom_data(date_time_hour_beginning,original_res_forecast_before_lockdown,residual_forecast,rsa_contracted_forecast,dispatchable_generation,residual_demand,rsa_contracted_demand,international_exports,international_imports,thermal_generation,nuclear_generation,eskom_gas_generation,eskom_ocgt_generation,hydro_water_generation,pumped_water_generation,ils_usage,manual_load_reduction,ios_excl_ils_and_mlr,dispatchable_ipp_ocgt,eskom_gas_sco,eskom_ocgt_sco,hydro_water_sco,pumped_water_sco,pumping,wind,pv,csp,other_re,total_re,wind_installed_capacity,pv_installed_capacity,csp_installed_capacity,other_re_installed_capacity,total_re_installed_capacity,installed_eskom_capacity,total_pclf,total_uclf,total_oclf,total_uclf_and_oclf,non_comm_sentout,drakensberg_gen_unit_hours,palmiet_gen_unit_hours,ingula_gen_unit_hours)
FROM '/Users/ds_learner_14/Documents/Eskom_Data_Analysis/ESK2033.csv'
DELIMITER ','
CSV HEADER
;


 * postgresql://postgres:***@localhost:5430/eskom_db
37704 rows affected.


[]

In [46]:
%%sql 
DROP TABLE IF EXISTS eskom_clean;
CREATE TABLE eskom_clean 
AS 
SELECT
    EXTRACT(YEAR FROM date_time_hour_beginning) as year,
    EXTRACT(MONTH FROM date_time_hour_beginning) as month,
    EXTRACT(DAY FROM date_time_hour_beginning) as day,
    date_time_hour_beginning,
    original_res_forecast_before_lockdown,
    residual_forecast,
    rsa_contracted_forecast,
    dispatchable_generation,
    residual_demand,
    rsa_contracted_demand,
    international_exports,
    international_imports,
    thermal_generation,
    nuclear_generation,
    eskom_gas_generation,
    eskom_ocgt_generation,
    hydro_water_generation,
    pumped_water_generation,
    ils_usage,
    manual_load_reduction,
    ios_excl_ils_and_mlr,
    dispatchable_ipp_ocgt,
    eskom_gas_sco,
    eskom_ocgt_sco,
    hydro_water_sco,
    pumped_water_sco,
    pumping,
    wind,
    pv,
    csp,
    other_re,
    total_re,
    wind_installed_capacity,
    pv_installed_capacity,
    csp_installed_capacity,
    other_re_installed_capacity,
    total_re_installed_capacity,
    installed_eskom_capacity,
    total_pclf,
    total_uclf,
    total_oclf,
    total_uclf_and_oclf,
    non_comm_sentout,
    drakensberg_gen_unit_hours,
    palmiet_gen_unit_hours,
    ingula_gen_unit_hours

FROM eskom_data;


 * postgresql://postgres:***@localhost:5430/eskom_db
Done.
75408 rows affected.


[]

# Numbers of rows and columns
 count rows=42824 estimated before clean after 37704 removedforecast dates added average for original res before lockdown 
count cols=41


look at our variables

In [29]:
all = %sql SELECT * FROM eskom_clean LIMIT 5
all

 * postgresql://postgres:***@localhost:5430/eskom_db
5 rows affected.


[(Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19904.967'), Decimal('20367.066'), Decimal('20237'), Decimal('20237'), Decimal('20722.058'), Decimal('1215.902'), Decimal('1120'), Decimal('19444'), Decimal('931'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1614'), Decimal('454.964'), Decimal('0'), Decimal('17.555'), Decimal('12.539'), Decimal('485.058'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3987.472'), Decimal('8028.71'), Decimal('275.907'), Decimal('8304'), Decimal('617'), Decimal('383'), Decimal('81.8'), Decimal('36.9'), Decimal('30.41')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19553.899'), Decimal('19988.733'), Decimal('19744'), Decimal('19744'), Decimal('20188.493'), Decimal('1203.474'), Decimal('1106'), Decimal('19297'), Decimal('930'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1945'), Decimal('392.958'), Decimal('0'), Decimal('39.131'), Decimal('12.404'), Decimal('444.493'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3987.472'), Decimal('7727.302'), Decimal('244.907'), Decimal('7972'), Decimal('209'), Decimal('388'), Decimal('83'), Decimal('38.5'), Decimal('32.85')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19314.284'), Decimal('19731.239'), Decimal('19631'), Decimal('19631'), Decimal('20019.603'), Decimal('1177.571'), Decimal('1117'), Decimal('19165'), Decimal('931'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1938'), Decimal('337.358'), Decimal('0'), Decimal('39.163'), Decimal('12.082'), Decimal('388.603'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3987.472'), Decimal('7704.704'), Decimal('193.727'), Decimal('7898'), Decimal('431'), Decimal('388'), Decimal('83.8'), Decimal('40.3'), Decimal('35.6')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19342.679'), Decimal('19753.554'), Decimal('19731'), Decimal('19731'), Decimal('20079.454'), Decimal('1184.312'), Decimal('1118'), Decimal('19279'), Decimal('930'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1952'), Decimal('297.234'), Decimal('0'), Decimal('38.942'), Decimal('12.278'), Decimal('348.454'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3990.072'), Decimal('7702.868'), Decimal('187'), Decimal('7889'), Decimal('868'), Decimal('389'), Decimal('85'), Decimal('42'), Decimal('37.76')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19538.89'), Decimal('19988.365'), Decimal('19890'), Decimal('19890'), Decimal('20237.49'), Decimal('1197.271'), Decimal('1108'), Decimal('19369'), Decimal('930'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1873'), Decimal('309.978'), Decimal('0'), Decimal('25.024'), Decimal('12.488'), Decimal('347.49'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3990.472'), Decimal('7685.115'), Decimal('187'), Decimal('7872'), Decimal('115'), Decimal('385'), Decimal('85.

## Comparing

Wind capacity

PV Capacity

CSP Capacity

Other RE Capacity

Eskom Capacity

Total Capacity

In [41]:
avg_capacity = %sql SELECT year AS "year",AVG(wind_installed_capacity) AS "wind",avg(pv_installed_capacity) AS "pv", AVG(csp_installed_capacity) AS "csp", AVG(other_re_installed_capacity) AS "other_re", AVG(installed_eskom_capacity) AS "installed_eskom_capacity" ,AVG(total_re_installed_capacity) as "total_re" FROM eskom_clean WHERE year >= '2018' and year <= '2022' GROUP BY year ORDER BY year
avg_capacity

 * postgresql://postgres:***@localhost:5430/eskom_db
5 rows affected.


[(Decimal('2018'), Decimal('1474.1900000000000000'), Decimal('311.6363636363636364'), Decimal('21.7800000000000000'), Decimal('3886.6987272727272727'), Decimal('4198.7305898484848485'), Decimal('44343.047272727273')),
 (Decimal('2019'), Decimal('1474.1900000000000000'), Decimal('491.7808219178082192'), Decimal('21.7800000000000000'), Decimal('4067.5108219178082192'), Decimal('4329.1675918949771689'), Decimal('44178.821917808219')),
 (Decimal('2020'), Decimal('1780.3017486338797814'), Decimal('500.0000000000000000'), Decimal('21.7800000000000000'), Decimal('4439.6280874316939891'), Decimal('5063.8692882513661202'), Decimal('45083.792349726776')),
 (Decimal('2021'), Decimal('2194.0434246575342466'), Decimal('500.0000000000000000'), Decimal('25.0073972602739726'), Decimal('5380.9399726027397260'), Decimal('5020.8706051369863014'), Decimal('46464.608219178082')),
 (Decimal('2022'), Decimal('2212.0900000000000000'), Decimal('500.0000000000000000'), Decimal('41.8300000000000000'), Decimal('5893.4680000000000000'), Decimal('4811.5026316666666667'), Decimal('47084.100000000000'))]

In [38]:
fig = px.line(avg_capacity, x="year", y =["wind","pv","csp"],title="Average of wind,pv and csp Capacity")
fig.show()

In [43]:
fig1 = px.line(avg_capacity, x="year", y =["installed_eskom_capacity", "other_re"],title="Average of eskom and other_re")
fig1.show()